### PR TITLE
fix: maintain query parameters

### PR DESCRIPTION
**Pull request by sakulstra**
- bump visx/tooltip: resolves strict mode error
- maintain current query parameters when adding a new query parameter: issue get's visible when e.g. directly going to https://app.aave.com/reserve-overview/?underlyingAsset=0x7f5c764cbc14f9669b88837ca1490cca17c31607&marketName=proto_optimism_v3
- optimize boot: only set the market from localStorage when it's different to the server rendered one
- "optimize" getAssetCapData: When you declare the function inside, every time the component is rendered the function will be defined again. That said, with this fix applied, the initial page load for this page is still crappy as it returns `null` and does not even render a layout. I don't understand why a context is used here, but also didn't have time to further clean this up.

### CODE REVIEW

File: src/store/utils/queryParams.ts [hunk 1]
```diff
@@ -2,7 +2,9 @@ import Router from 'next/router';
 
 export const setQueryParameter = (key: string, value: string) => {
   if (typeof window !== 'undefined') {
-    Router.push({ query: { [key]: value } }, undefined, { shallow: true });
+    Router.push({ query: { ...getAllQueryParameters(), [key]: value } }, undefined, {
```
Comment (sakulstra) [inline src/store/utils/queryParams.ts:5]: as this runs on app startup the Router.state.query might not be ready yet so using native apis to fetch current parameters

**Review by grothem — DISMISSED** [review]

**Comment by sakulstra** [comment]
@grothem plz don't merge yet, might have found another issue

File: src/store/utils/queryParams.ts (hunk 1 above)
Comment (grothem) [reply src/store/utils/queryParams.ts:6]: You left a `console.log` in

**Review by defispartan — APPROVED** [review]

**Comment by drewcook** [comment]
@sakulstra Cool, please add the `Ready for Dev Review` label when it's all cleaned up and ready for a final review :)

File: src/store/utils/queryParams.ts [hunk 2]
```diff
@@ -2,7 +2,10 @@ import Router from 'next/router';
 
 export const setQueryParameter = (key: string, value: string) => {
   if (typeof window !== 'undefined') {
-    Router.push({ query: { [key]: value } }, undefined, { shallow: true });
+    console.log('here');
```
Comment (sakulstra) [inline src/store/utils/queryParams.ts:5]:
```suggestion
```

**Review by drewcook — APPROVED** [review]
Good catch, LGTM 👍

**Review by grothem — APPROVED** [review]

**Pull request merged**